docs/phase_0.md

```diff
@@ -363,339 +363,1026 @@ class SignedVoluntaryExit(Container):
+    * `slot` -- tracks the slot of the containing state. Note that this isn't necessarily the same as the slot according to the local wall-clock. Remember that time is divided into slots of fixed length (`SECONDS_PER_SLOT`) in which actions occur and state transitions happen
     * `fork` -- a mechanism for handling forking (upgrading) the beacon chain. The main purpose here is to handle versioning of signatures and handle objects of different signatures across fork boundaries
 * History
-    * `latest_block_header` -- cache of the latest block header seen in the chain defining this state. During the slot transition of the block, the header is stored _without_ the real state root but instead with a stub of `Root()` (empty `0x00` bytes). At the start of the _next slot_ transition before anything has been modified within state, the state root is calculated and added to the `latest_block_header`. This is done to eliminate the circular dependency of the state root being embedded in the block header while still allowing state roots to be referenced within the beacon state
-    * `block_roots` -- per-slot store of the recent block roots. The block root for a slot is added at the start of the next slot to avoid the circular dependency due to the state root being embedded in the block. For slots that are skipped (no block in the chain for the given slot), the most recent block root in the chain prior to the current slot is stored for the skipped slot. When validators attest to a given slot, they use this store of block roots as an information source to cast their vote
-    * `state_roots` -- per-slot store of the recent state roots. The state root for a slot is stored at the start of the next slot to avoid a circular dependency
-    * `historical_roots` -- a double batch merkle accumulator of the latest block and state roots defined by [`HistoricalBatch`](https://github.com/ethereum/eth2.0-specs/blob/v0.10.0/specs/phase0/beacon-chain.md#historicalbatch) to make historic merkle proofs against. Note that although this field grows unbounded, it grows at less than 10 KB per year
+    * `latest_block_header` -- a cache of the latest block header seen in the chain defining this state. During the slot transition of the block, the state `Root` (a 32 byte vector) in the header is stored with an empty placeholder (`0x00..00`) instead of the real state root. At the start of the _next slot_ transition -- before anything has been modified within state -- the real state root is calculated and added to the `latest_block_header`. We do this to prevent the state root from being embedded in the block header while the block header is part of the state -- the state root is a hash and we can’t embed an object's hash into itself without changing that object's hash (what we call a circular dependency). The placeholder allows us to reference the state root within the beacon state and get around this circular dependency.
```

Comment:
```suggestion
    * `latest_block_header` -- a cache of the latest block header seen in the chain defining this state. During the slot transition of the block, the state `Root` (a 32 byte vector) in the header is stored with an empty placeholder (`0x00..00`) instead of the real state root. At the start of the _next slot_ transition -- before anything has been modified within state -- the real state root of the previous slot is calculated and added to the `latest_block_header`. We do this to prevent the state root from being embedded in the block header while the block header is part of the same slot's state -- the state root is a hash and we can’t embed an object's hash into itself without changing that object's hash (what we call a circular dependency). The placeholder helps us get around this circular dependency by allowing us to insert the state root in the header within the beacon state at a subsequent slot.  
```